Temporary shortcuts + Flip notify, honor change when leaving play, reduce cost prompt

Kertanos opened the pull request.

Kertanos committed:
update

Kertanos committed:
Merge remote-tracking branch 'refs/remotes/blakewedwards/master'

Kertanos committed:
Merge remote-tracking branch 'refs/remotes/blakewedwards/master'

# Conflicts:
#	o8g/Sets/Core Set/set.xml

Kertanos committed:
Temporary Shortcuts

Kertanos committed:
Flip notify, honor change when leaving play, reduce cost prompt

Kertanos committed:
Revert "Temporary Shortcuts"

This reverts commit 99546b8635a0724b5fe2264e6cff563bf8c641f2.

Kertanos committed:
Temporary shortcuts

blakewedwards commented on o8g/scripts/actions.py:
```diff
@@ -337,6 +337,7 @@ def flip(card, x=0, y=0):
     card.alternate = ALTERNATE_POLITICAL if not card.alternate else ''
     return
   card.isFaceUp = not card.isFaceUp
+  notify("{} flips {}.".format(me,card))
```
Realize what the problem is seeing the function, the card.isFaceUp toggle will have the generic octgn notify so it's ok, but the if check outputs nothing. Could you move the notify to just above the return and phrase it like ```notify("{} turns '{}' to {}".format(me, card, card.alternate if card.alternate else 'Military'))```

blakewedwards replied:
Would also be good to add a mute() to just the if section in case octgn adds messages for those commands

Kertanos replied:
Yeah i misunderstood your trello card, thought the isFaceUp function didn't show anything ^^
i'll correct that

blakewedwards commented on o8g/scripts/actions.py:
```diff
@@ -362,6 +363,12 @@ def has_deck(card, x=0, y=0):
   if pile is not None:
+    if card.markers[HONORED]:
+      me.honor+=1
+      notify("{} gains 1 honor from {} honored status.".format(me, card))
```
I haven't muted this function so the notify isn't needed at the moment. In general I've tried to use the generic messages since they describe the actions in the way you'd describe a player normally interacting with the actual cards. What you'll see here if the notify is removed is ```Intolerance sets Intolerance's Honor counter to 12 (+1)
Intolerance moves 'Doomed Shugenja' to Intolerance's Dynasty Discard``` which is the correct ordering due to the honor change being an interrupt

Kertanos replied:
I wanted to give a bit more information about where the gain or loss of honor was coming from.
Doesn't really matter when you discard only one card, but when you select and discard many, like you would in fate phase, it can be usefull.
But maybe it's too much yeah

blakewedwards commented on o8g/definition.xml:
```diff
@@ -65,27 +65,27 @@
-    <groupaction menu="Claim ring" execute="claim_ring" showIf="setup_not_required"/>
-    <groupaction menu="End turn" execute="end_turn" showIf="setup_not_required"/>
-    <groupaction menu="Flip coin" execute="flip_coin" showIf="setup_not_required"/>
+    <groupaction menu="Set up" execute="setup" showIf="setup_required" shortcut="CTRL+SHIFT+S" />
```
Looks good, I didn't realize that shortcuts got around showIf.. too bad but good to know. I'm a bit curious how you chose some of the shortcuts since they don't match the letters in the word. Does it match the conquest versions or the letters for the word in a different language?

Kertanos replied:
well there is some actions that share letters, so i had to find something else.
Some are also coming from conquest yeah, like ALT+D for pass, was the shortcuts for "i'm done"
Like i said, it's temporary, we can find a better list of shortcuts i think ^^

blakewedwards commented on o8g/scripts/actions.py:
```diff
@@ -395,12 +402,24 @@ def random_discard_from(group):
 def can_play(card, x=0, y=0):
   return unpack(card, lambda c: c.isFaceUp and (c.type == TYPE_CHARACTER or c.type == TYPE_EVENT or c.type == TYPE_ATTACHMENT))
 
+def prompt_reduce_cost(cost):
```
Looks great

Kertanos committed:
update from blake master

Kertanos committed:
corrections from feedback

Kertanos committed:
Cards are now face up by default

Kertanos commented:
There is a problem with flipping rings and imperial favor.
It can't seems to find the card properties once flipped to notify.
it's also strangely retroactive.
Meaning when i try to notify card or card.name, it shows nothing.
Worse, the name then shows on the previous message when i flip it back.
Then disappear again when you reflip it...

So for now i decided to cancel any changes, and not notify anything yet.
I'll investigate.

Kertanos committed:
Proxies update(Phoenix and neutral holding missing), new role proxies, FP token

Kertanos committed:
Lots of stuff !

Implemented mulligan
added initiative choice on setup + bonus fate for second player
set default click on honor dial to "select bid"
multiple card placement arrangement so that it all feels better for the
eyes and is easier to play.
Also added player summary
and the little icing on the cake, a better looking dishonored token
(still not perfect :(  )

blakewedwards commented on o8g/Sets/Core Set/set.xml:
```diff
@@ -130,11 +130,11 @@
       <property name="Type" value="Stronghold"/>
       <property name="Unique" value="Unique"/>
       <property name="Traits" value="City."/>
-      <property name="Text" value="???"/>
```
Hmm I'm not sure why these changes are showing up, they should already exist and don't need to be merged

blakewedwards commented on o8g/Sets/Core Set/set.xml:
```diff
@@ -2327,69 +2472,79 @@
       <property name="Type" value="Role"/>
       <property name="Traits" value="Keeper. Air."/>
       <property name="Text" value="Increase your deckbuilding influence value by 3.&#10;Reaction: After you win an [Air] conflict as the defending player - gain 1 fate."/>
+      <property name="Ring" value="Air"/>
```
The set data is automatically generated so anything manually put here will get overwritten. I think it might be simpler to generate the proxy based on another property of the role until we get better images, like name, traits, or card number

Kertanos replied:
Ok i'll try something like that

blakewedwards commented on o8g/definition.xml:
```diff
@@ -64,32 +64,33 @@
-    <groupaction menu="End turn" execute="end_turn" showIf="setup_not_required"/>
-    <groupaction menu="Flip coin" execute="flip_coin" showIf="setup_not_required"/>
+    ordered="False"
+    visibility="undefined">
```
Nice

blakewedwards commented on o8g/definition.xml:
```diff
@@ -64,32 +64,33 @@
+    <cardaction menu="Refill" execute="refill" showIf="is_province" shortcut="CTRL+R" />
+    <cardaction menu="Replace" execute="replace" showIf="can_replace" shortcut="CTRL+T" />
     <cardaction menu="Select bid" execute="select_bid" showIf="is_honor_dial"/>
-    <cardaction menu="Move to bottom of deck" execute="move_to_deck_bottom" showIf="has_deck"/>
```
I didn't realize this already existed in the generic move-to menu.

Kertanos replied:
and you can also just drag and drop the cards to the bottom of the deck (there is a little arrow that shows you you're moving to bottom instead of top)

blakewedwards commented:
Looks great, only things I'm not sure about are why some of the set diffs are appearing in the PR and the new fields in the set data.

Kertanos committed:
fixed role proxies image to depends on elemental Traits

Kertanos commented:
There you go, Role elemental rings now depends from Traits instead or Rings.
Not sure what i can do about the weird set.xml "changes that are not changes because it's already in both version"
